### PR TITLE
Fix describe with injected gateway not working

### PR DESCRIPTION
**Pull request by hanxiaop**
**Please provide a description of this PR:**
Originally it uses the service label selector, which injected gateway doesn't have. This PR revises it to use pod selector.

### CODE REVIEW

**Comment by hanxiaop** [comment]
Resolve https://github.com/istio/istio/issues/46160#issuecomment-1649402195

File: istioctl/pkg/describe/describe.go [hunk 1]
```diff
@@ -906,13 +906,18 @@ func printIngressInfo(
...
-		LabelSelector: "istio=ingressgateway",
-	})
+	ingressSvcs, err := kubeClient.CoreV1().Services(istioNamespace).List(context.TODO(), metav1.ListOptions{})
 	if err != nil {
 		return multierror.Prefix(err, "Could not find ingress gateway service")
 	}
-	if len(ingressSvcs.Items) == 0 {
+	filteredIngressSvcs := []corev1.Service{}
+	for _, svc := range ingressSvcs.Items {
+		svc := svc
```
Comment (ymesika) [inline istioctl/pkg/describe/describe.go:915]: What's the purpose of this re-assigment?

Comment (hanxiaop) [reply istioctl/pkg/describe/describe.go:915]: Thanks, forgot deleting this, it's not necessary here.

**Review by ymesika — APPROVED** [review]
lgtm

**Review by GregHanson — APPROVED** [review]

**Pull request merged**